### ISSUE TITLE
factorint(pow(10,38)-1)

### ISSUE THREAD

**Issue by BaltiBoix** [sympy/sympy#19683]
Factorint of pow(10,n)-1 takes forever for n=38. No problem with n = 37, 39, 40. 

From https://www.alpertron.com.ar/ECMC.HTM:

99 999999 999999 999999 999999 999999 999999 (38 dígitos) = 32 × 11 × 909090 909090 909091 × 1 111111 111111 111111

**Comment by Yathartha22** [comment]
@abhinav28071999 

#19436 implements `ecm`. Maybe we need to use `ecm` instead of `factorint` for larger digits.

**Comment by jksuom** [comment]
`factorint` is a wrapper function that currently tries Pollard's p - 1 and rho methods after removing small prime factors by trial division. Elliptic curve method and quadratic sieve are extensions of Pollard's two methods that are more expensive but can be applied to bigger numbers. `factorint` should use them automatically if more elementary methods fail. `factorint` should be the only function that a typical user would need, even for big numbers like the one in the title.

**Comment by abhinav28071999** [comment]
> `factorint` is a wrapper function that currently tries Pollard's p - 1 and rho methods after removing small prime factors by trial division. Elliptic curve method and quadratic sieve are extensions of Pollard's two methods that are more expensive but can be applied to bigger numbers. `factorint` should use them automatically if more elementary methods fail. `factorint` should be the only function that a typical user would need, even for big numbers like the one in the title.

i agree. Ecm and QS needs to be integrated in factorint. I  will make  suitable changes to pollard methods. When ph methods fail them we use ecm then if it also fail then try qs.

**Comment by sweeneyde** [comment]
Are there any plans to still integrate QS into factorint?
